NuRadioReco/modules/io/RNO_G/readRNOGDataMattak.py

```diff
@@ -724,29 +728,20 @@ def run(self):
         for dataset in self._datasets:
             dataset.setEntries((0, dataset.N()))
 
-            # read all event infos of the entier dataset (= run)
-            event_infos = dataset.eventInfo()
-            wfs = None
-
-            for idx, evtinfo in enumerate(event_infos):  # returns a list
+            # read all event infos of the entire dataset (= run)
+            for evtinfo, wf in dataset.iterate(start = 0, stop = dataset.N(),
+                                               calibrated = self._read_calibrated_data,
+                                               selectors = self._selectors,
+                                               max_entries_in_mem = self._max_in_mem):
                 event_idx += 1
 
-                self.logger.debug(f"Processing event number {event_idx} out of total {self._n_events_total}")
-                t0 = time.time()
-
-                if self._filter_event(evtinfo, event_idx):
-                    continue
+                self.logger.debug(f"Processing (filtered) event number {event_idx} out of (unfiltered) total {self._n_events_total}")
+                t0 =time.time()
 
                 if not self._check_for_valid_information_in_event_info(evtinfo):
-                    continue
-
-                # Just read wfs if necessary
-                if wfs is None:
-                    wfs = dataset.wfs()
-
-                waveforms_of_event = wfs[idx]
+                    continue 
```

Comment:
Can you try passing this function as selector? The the `_filter_event` function also. That should speed things up

Reply:
Ok, but I had to invert the logic in the _filter_event function to mimic the selectors logic i.e. I changed it to return True if the event is to be read and False if the event should be skipped.